kythe/go/util/tools/write_extra_action/write_extra_action.go

```diff
@@ -80,3 +86,17 @@ func main() {
 		log.Fatal(err)
 	}
 }
+
+func populateCppCompileInfo(cc *xapb.CppCompileInfo) {
```

Comment:
Can we add a flag to pass in this as another textproto?  Using the non-flag arguments seems too specialized for this.

Reply:
We cannot. That is the entire motivation for this change.

Reply:
Why can't the caller structure the arguments into a textproto if it already has them?

Reply:
This only uses the trailing non-flag arguments if there is a CppCompileInfo extension provided and only after an explicit `--` delimiter, in theory allowing for some other interpretation of those arguments for a different extension.  I don't anticipate such a need as only "legacy" rules have custom protocol buffers at all and no more are likely to be added.  Extra actions being removed entirely seems more likely than new ExtraActionInfo extensions.

Reply:
The caller is Bazel. It does not provide access to the stringified command line arguments, only an opaque Args object.